Allow trailing spaces when parsing SVGLengthValue

nullhook opened the pull request:
#### 580c901d87d7f4ff32e3494744ecb35a41334c78
<pre>
Allow trailing spaces when parsing SVGLengthValue
<a href="https://bugs.webkit.org/show_bug.cgi?id=297819">https://bugs.webkit.org/show_bug.cgi?id=297819</a>
<a href="https://rdar.apple.com/158999226">rdar://158999226</a>

Reviewed by Tim Nguyen.

We allow trailing spaces when parsing SVGLengthValue.
Essentially, we let the CSS parsing machinery handle trailing spaces.

LayoutTests/svg/css/parse-height-expected.txt:
LayoutTests/svg/css/parse-length-expected.txt:
LayoutTests/svg/custom/invalid-length-units-expected.txt:
LayoutTests/svg/custom/invalid-length-units.html:
Source/WebCore/svg/SVGLengthValue.cpp:
(WebCore::SVGLengthValue::setValueAsString):

Canonical link: <a href="https://commits.webkit.org/299098@main">https://commits.webkit.org/299098@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b69652ff907cb4703d5ea9cf7ddc72da62c1a3af

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117822 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37500 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28130 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123960 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69846 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c95f3628-280f-4170-a53c-39b1394a1e58) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119700 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38192 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46082 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89415 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6e6e2fac-cc91-4dd6-95fb-164d9fd011bc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120774 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30409 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105644 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69908 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/48f8cb55-5373-4ebf-bc29-2f44e43ad7e7) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29473 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23760 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67622 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99829 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23939 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127049 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44725 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33673 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98078 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45085 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101869 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97866 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24900 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43253 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21224 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41114 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44597 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50271 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44056 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47402 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45746 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/svg/SVGLengthValue.cpp:
```diff
@@ -343,12 +343,6 @@ ExceptionOr<void> SVGLengthValue::setValueAsString(StringView string)
 
-    // FIXME: Allow leading and trailing whitespace in SVG attributes
-    // using <integer>, <angle>, <number>, <length>, and <percentage>
-    // rdar://115963075
```
We probably could re-use the other bug

nullhook replied:
this PR resolves a subset of the "other bug". it's basically missing `<angle>`

nt1m commented on LayoutTests/svg/custom/invalid-length-units.html:
```diff
@@ -26,8 +26,8 @@
-trySettingLength("10px ", "'0'");
-trySettingLength("10% ", "'0'");
+trySettingLength("10px ", "'10px'");
+trySettingLength("10% ", "'10%'");
```
I think it would be good to move these at the end with the other valid testcases

Ahmad-S792 replied:
Can we just replace it with from `source.chromium.org`, which uses 'testharness.js'?

https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/svg/custom/invalid-length-units.html;l=1;bpv=1;bpt=0

nullhook committed:
Allow trailing spaces when parsing SVGLengthValue
https://bugs.webkit.org/show_bug.cgi?id=297819
rdar://158999226

Reviewed by Tim Nguyen.

We allow trailing spaces when parsing SVGLengthValue.
Essentially, we let the CSS parsing machinery handle trailing spaces.

LayoutTests/svg/css/parse-height-expected.txt:
LayoutTests/svg/css/parse-length-expected.txt:
LayoutTests/svg/custom/invalid-length-units-expected.txt:
LayoutTests/svg/custom/invalid-length-units.html:
Source/WebCore/svg/SVGLengthValue.cpp:
(WebCore::SVGLengthValue::setValueAsString):

Canonical link: https://commits.webkit.org/299098@main
